tests/kernel/mutex/mutex_api/src/test_mutex_apis.c

```diff
@@ -91,7 +108,99 @@ static void tmutex_test_lock_unlock(struct k_mutex *pmutex)
 	k_mutex_unlock(pmutex);
 }
 
+static void tThread_T1_priority_inheritance(void *p1, void *p2, void *p3)
+{
+	/* t1 will get mutex first */
+	zassert_true(k_mutex_lock((struct k_mutex *)p1, K_FOREVER) == 0,
+		      "access locked resource from spawn thread T1");
+
+	/* record its original priority */
+	int priority_origin = k_thread_priority_get((k_tid_t)p2);
+
+	TC_PRINT("T1 is going to enter mutex: origin priority=%d\n",
+		priority_origin);
+
+	/* wait for a time period to see if priority inheritance happended */
+	k_sleep(K_MSEC(500));
+
+	int priority = k_thread_priority_get((k_tid_t)p2);
+
+	TC_PRINT("access resource from thread T1: priority=%d\n", priority);
+
+	zassert_true(priority <= priority_origin,
+		      "priority inheritance not happened!");
```

Comment:
Would be better to test for equality with the priority of the thread that is known to be waiting.  The behavioral requirement isn't just that the priority increase, it's that the mutex-holding thread behave to the scheduler as if it "was" the higher priority thread that is waiting.

Reply:
Yes, I think you are totally right. The priority should increase to the same with waiter thread. I rework it and add 3rd thread in order to improve the branch coverage slightly to 73.1%. Thanks for your advice!